components/molecules/HighlightInput/highlight-input-form.tsx

```diff
@@ -0,0 +1,92 @@
+import Button from "components/atoms/Button/button";
+import { ChangeEvent, useEffect, useRef, useState } from "react";
+
+const HighlightInputForm = (): JSX.Element => {
+  const [isDivFocused, setIsDivFocused] = useState<boolean>(false);
+  const textAreaRef = useRef<HTMLTextAreaElement>(null);
+  const [bodyText, setBodyText] = useState<string>("");
+  const [row, setRow] = useState<number>(1);
+  const [title, setTitle] = useState<string>("");
+  const ref = useRef<HTMLFormElement>(null);
+  let rowLomit = 5;
+  let messageLastScrollHeight = textAreaRef.current ? textAreaRef.current?.scrollHeight : 50;
+
+  useEffect(() => {
+    const checkIfClickedOutside = (e: any) => {
+      // If the menu is open and the clicked target is not within the menu,
+      // then close the menu
+      if (isDivFocused && ref.current && !ref.current.contains(e.target)) {
+        setIsDivFocused(false);
+      }
+    };
+    document.addEventListener("mousedown", checkIfClickedOutside);
+
+    return () => {
+      // Cleanup the event listener
+      document.removeEventListener("mousedown", checkIfClickedOutside);
+    };
+  }, [isDivFocused, bodyText]);
+
+  const handleTextAreaInputChange = (e: ChangeEvent<HTMLTextAreaElement>) => {
+    setBodyText(e.target.value);
+    if (row < rowLomit && textAreaRef.current && textAreaRef.current?.scrollHeight > messageLastScrollHeight) {
+      setRow((prev) => prev + 1);
+    } else if (row > 1 && textAreaRef.current && textAreaRef.current?.scrollHeight < messageLastScrollHeight) {
+      setRow((prev) => prev--);
+    }
+
+    if (!bodyText) setRow(1);
+    messageLastScrollHeight = textAreaRef.current?.scrollHeight || 65;
+  };
+
+  // Handle submit highlights
+  const handlePostHighlight = () => {
+    // Trigger api call to post highlight
+    setBodyText("");
+    setTitle("");
+  };
+
+  return (
+    <form
+      onSubmit={(e) => {
```

Comment:
Let's move the logic into the callback function

```suggestion
      onSubmit={handlePostHighlight}
```